### PR TITLE
[ACIX-424] Add linter for job owners

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

> [!NOTE]
> The logic to get which jobs have to be lint has been refactored

This adds a linter that verifies that each job has an owner

### CODE REVIEW

File: .gitlab/.pre/gitlab_configuration.yml [hunk 1]
```diff
@@ -62,6 +62,8 @@ lint_gitlab_ci_jobs_needs_rules:
     - changes:
         paths:
           - .gitlab/.ci-linters.yml
+          - .gitlab/JOBOWNERS
```
Comment (KevinFairise2) [inline .gitlab/.pre/gitlab_configuration.yml:65]: Do we really need that? I think this is already included in `.on_gitlab_changes` isn'it?

**Review by KevinFairise2 — APPROVED** [review]

**Review by amenasria — APPROVED** [review]
LGTM as long as the CI passes

File: .gitlab/.pre/gitlab_configuration.yml (hunk 1 above)
Comment (CelianR) [reply .gitlab/.pre/gitlab_configuration.yml]: Indeed, nice catch

**Pull request merged**